### DISCUSSION TITLE
What's the structure of the Climate Summary product text?

### DISCUSSION THREAD

**Discussion by fury88** [weather-gov/api#766] [Q&A]
Hi NWS, et al.

I'm currently trying to get the daily climate summary for a project I'm working and while I can retrieve it fine, it doesn't seem to be in any kind of tabular or json format. It's formatted nicely if you want to print that text to the screen as is but creating a table out of it is darn near impossible without a lot of python parsing. Even then its not easy. Am I missing something or does anyone have any suggestions on doing this?

Here is an example of the printout straight from the text result:

```
WEATHER ITEM   OBSERVED TIME   RECORD YEAR NORMAL DEPARTURE LAST      
                VALUE   (LST)  VALUE       VALUE  FROM      YEAR     
                                                  NORMAL           
...................................................................
TEMPERATURE (F)                                                          
 YESTERDAY                                                           
  MAXIMUM         86    403 PM  98    1895  81      5       84       
  MINIMUM         64    626 AM  41    1962  61      3       61       
  AVERAGE         75                        71      4       73     

PRECIPITATION (IN)                                                    
  YESTERDAY        0.82          1.62 1979   0.14   0.68     0.00     
  MONTH TO DATE    8.13                      3.72   4.41     2.67     
  SINCE SEP 1      8.13                      3.72   4.41     2.67     
  SINCE JAN 1     44.81                     34.56  10.25    31.47     

DEGREE DAYS                                                           
 HEATING                                                              
  YESTERDAY        0                         1     -1        0        
  MONTH TO DATE    0                         2     -2        0        
  SINCE SEP 1      0                         2     -2        0        
  SINCE JUL 1      0                         2     -2        0        

 COOLING                                                              
  YESTERDAY       10                         7      3        8        
  MONTH TO DATE  175                       194    -19      245        
  SINCE SEP 1    175                       194    -19      245        
  SINCE JAN 1   2025                      1690    335     1816        
..................................................................
```

But clear while it IS in a fixed format in some cases it's not consistent all the way through.

Thanks!

-MF

**Comment by JJKraw** [comment]
NWS text products were created and continue to be produced to be read by humans. The API just provides another way they are available. A considerable effort would be needed to transition to something more machine-friendly.

**Reply by fury88** [reply]
> NWS text products were created and continue to be produced to be read by humans. The API just provides another way they are available. A considerable effort would be needed to transition to something more machine-friendly.

Yes. As I'm figuring that out now. While I understand, it seems strange to produce a human readable format from an API. The end result should leave it up to the consumer.

**Reply by JJKraw** [reply]
The point is that the products predate the API and are meant for human consumption. A good number of products (if not most) are not tabular; they're just text conveying a message. See Area Forecast Discussion for a good example. You just happened to pick something that displays data in a tabular format. As a first step for the API, I assume the choice was to just make all of the products available as-is and not require the forecast offices to have to change anything regarding how the products are published.

**Reply by fury88** [reply]
Yes, even the Climate Summary is not truly tabular. I can massage it to get into a DataFrame but its not easy. I mean the api comes in json format anyway so I don't think it would be much of an effort to add a few more key/value pairs. As far as entering it, not much would have to change either. Anyway, I get there I budgetary concerns as well. As for synopsis. That was a little easier to parse out because it's pretty much a paragraph of text in between some metadata. Maybe I'll take this request over to 2.0 to see if it may be on their radar.

**Reply by akrherz** [reply]
Apologies for advertising other APIs, but generally [ACIS](https://www.rcc-acis.org/docs_webservices.html) has most of this data.  My project does have [JSON](https://mesonet.agron.iastate.edu/json/cli.py?help)/[geojson](https://mesonet.agron.iastate.edu/geojson/cli.py?help) services with atomic CLI data.  My junky python [CLI parser](https://github.com/akrherz/pyIEM/blob/main/src/pyiem/nws/products/cli.py) perhaps can provide some inspiration. FWIW

**Reply by dwhitemv25** [reply]
Machine-readable climate summaries can be retrieved from [NCEI](https://www.ncei.noaa.gov).
